src/Operations.jl

```diff
@@ -1148,7 +1148,8 @@ function update_package_add(ctx::Context, pkg::PackageSpec, entry::PackageEntry,
             println(ctx.io, "`$(pkg.name)` is pinned at `v$(entry.version)`: maintaining pinned version")
         end
         return PackageSpec(; uuid=pkg.uuid, name=pkg.name, pinned=true,
-                           version=entry.version, tree_hash=entry.tree_hash)
+                           version=entry.version, tree_hash=entry.tree_hash,
```

Comment:
Is it really possible to pin everything here? What if I update the version in the Project.toml file in my local path? Perhaps if you pin a package that tracks a path it should only be the path that is pinned?

Reply:
If you change the version in the `dev`ed path (**and pinned pkg**), then `Pkg.add("Example")` results in:
```
`Example` is pinned at `v0.5.3`: maintaining pinned version
   Resolving package versions...
    Updating `~/.julia/environments/v1.7/Project.toml`
  [7876af07] ~ Example v0.5.3 `[...]/Example.jl` ⚲ ⇒ v0.5.8 `[...]/Example.jl` ⚲
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [7876af07] ~ Example v0.5.3 `[...]/Example.jl` ⚲ ⇒ v0.5.8 `[...]/Example.jl` ⚲
```
I guess it is just consistent with `develop`, although the version is not `pinned` per se.